File: Test/AtCoderLibrary.Benchmark/Benchmark/SatisfiableTwoSat.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;
using AtCoder;

public static class SatisfiableTwoSat
{
    public static long Calc(int n)
    {
        n >>= 1;
        long ans = 0;
        var twoSat = new TwoSat(n);
        for (int i = 0; i < n; i++)
        {
            twoSat.AddClause(i, true, (i + 1) % n, false);
        }
        var ok = twoSat.Satisfiable();
        ans = twoSat.Answer()[0] ? 1 : -1;
        return ans;
    }
}

File: Test/AtCoderLibrary.Benchmark/Benchmark/C++/SatisfiableTwoSat.cpp
#include <atcoder/all>
#include <iostream>
#include <random>
#include <iostream>
#include <algorithm>
#include <vector>
#include <queue>
#include <set>
#include <map>
#include <math.h>
#include <unistd.h>
#include <unordered_set>
#include <unordered_map>
#define rep(i,n) for(int i=0;i<n;i++)
#define rep2(i,s,n) for(int i=s;i<n;i++)
#define repd(i,s,n,d) for(int i=s;i<n;i+=d)
#define repit(col) for(auto it = std::begin(col); it != std::end(col); ++it)

using namespace std;
using namespace atcoder;

using mint = modint998244353;
typedef long long ll;

int main() {
    int n = 1 << 24;;

    n >>= 4;
    long ans = 0;
    auto twoSat = two_sat(n);
    for (int i = 0; i < n; i++)
    {
        twoSat.add_clause(i, true, (i + 1) % n, false);
    }
    bool ok = twoSat.satisfiable();
    ans = twoSat.answer()[0] ? 1 : -1;
    printf("%lld\n", ans);
    return 0;
}
